app/javascript/components/navbar/AskQuestion.tsx

```diff
@@ -2,82 +2,128 @@ import React, { useState, useContext } from 'react';
 import {
   Form,
   Button,
-  Alert,
-  AlertProps,
+  Spinner,
 } from 'react-bootstrap';
-import Routes from '@hourglass/routes';
-import { getCSRFToken } from '@hourglass/helpers';
+import {
+  ProfQuestion,
+  ProfQuestionStatus,
+} from '@hourglass/types';
 import { RailsContext } from '@hourglass/context';
+import { DateTime } from 'luxon';
+import { ExhaustiveSwitchError } from '@hourglass/helpers';
+import { AiFillCheckCircle, AiFillCloseCircle } from 'react-icons/ai';
 
-async function submitQuestion(examID: number, question: string): Promise<boolean> {
-  const url = Routes.ask_question_exam_path(examID);
-  const res = await fetch(url, {
-    method: 'POST',
-    headers: {
-      'Content-Type': 'application/json',
-      'X-CSRF-Token': getCSRFToken(),
-    },
-    credentials: 'same-origin',
-    body: JSON.stringify({
-      message: {
-        body: question,
-      },
-    }),
-  });
-  const json = await (res.json() as Promise<{success: boolean}>);
-  return json.success;
+interface ShowStatusProps {
+  status: ProfQuestionStatus;
 }
 
-const AskQuestion: React.FC<{}> = () => {
-  const [val, setVal] = useState('');
-  const [error, setError] = useState('');
-  const [variant, setVariant] = useState<AlertProps['variant']>('danger');
-  const [showAlert, setShowAlert] = useState(false);
-  const [saving, setSaving] = useState(false);
-  const buttonText = saving ? 'Saving...' : 'Submit';
-  const { railsExam } = useContext(RailsContext);
+const ShowStatus: React.FC<ShowStatusProps> = (props) => {
+  const {
+    status,
+  } = props;
+  const size = '1.5em';
+  switch (status) {
+    case 'SENDING':
+      return (
+        <span className="text-info">
+          <Spinner
+            size="sm"
+            animation="border"
+          />
+        </span>
+      );
+    case 'FAILED':
+      return (
+        <span className="text-danger">
+          <AiFillCloseCircle
+            size={size}
+          />
+        </span>
+      );
+    case 'SENT':
+      return (
+        <span className="text-success">
+          <AiFillCheckCircle
```

Comment:
Maybe use the same cloud-check icon as for snapshots?  It occurred to me looking at this checkmark and x-mark that students might think they got an "Yes" or "No" answer from the professor, rather than "this message was successfully sent."  We should also add a title tooltip to these icons indicating what they mean.

Reply:
- [x] change icons
- [ ] tooltip

Reply:
Do you think a title is sufficient, or should we add a full-on `Tooltip`?

Reply:
It should probably be consistent everywhere, so if we're using Tooltips for the navbar status icons, we should use them everywhere, no?